app/lib/string_attribute_cleaner.rb

```diff
@@ -0,0 +1,32 @@
+# This module provides a method for generating a configurable before_validation
+# hook for cleaning up string attributes: all listed attributes will be
+# replaced with `nil` if they are blank.
+module StringAttributeCleaner
+  # This method specifies the list of attributes that should be cleaned before
+  # validation happens.
+  #
+  # @param attributes [Array<Symbol|String>]
+  #
+  # @return [void]
+  #
+  # @example Defining a list of attributes to be cleaned
+  #   include StringAttributeCleaner.for(:attribute1, :attribute2)
+  def self.for(*attributes)
```

Comment:
I really like this `for(*attributes)` API - one potential hiccup is the anonymous module could make method definition/source resolution less friendly in the console, but we're already using rails :)

Reply:
@djuber It's possible to make it a named module and if it ever becomes a problem I can do that later.